canonical/dotrun-image
# PoC: Support NPM and Yarn

## Done
This is a proof of concept for running NPM or Yarn depending on which lock file exists. This would enable the transition from NPM to Yarn if that approach was accepted. There some other minimal changes required (mostly renaming functions) if this was to land.

## How to QA
1. Check-out this branch
2. Build this image `docker build . --tag canonicalwebteam/dotrun-image:local`
3. Run this image `docker run -it -p 8004:8004 canonicalwebteam/dotrun-image:local /bin/bash`
4. Test an NPM project:
```bash
mkdir npmtest
cd npmtest
npm init # accept all defaults
npm install # this creates the package-lock.json
dotrun install # this should install with NPM
doturn test # this should run the script with npm
```
5. Test a Yarn project:
```bash
mkdir yarntest
cd yarntest
yarn init # accept all defaults
yarn install # this creates the yarn.lock
npm pkg set 'scripts.test'='echo "hello"' # this adds a script to run later
dotrun install # this should install with Yarn
doturn test # this should run the script with Yarn
```

`dotrun` should install the dependencies and run the project. Once you see this output:
```
Listening at: http://0.0.0.0:8004
```

Visit http://localhost:8004 and check that the website is running.

## Issue / Card
Fixes #
